Sources/Thread.swift

```diff
@@ -65,16 +65,28 @@ public final class Thread {
 
 // MARK: - Private
 
-private func ThreadPrivateMain(arg: UnsafeMutableRawPointer?) -> UnsafeMutableRawPointer? {
-    
-    let unmanaged = Unmanaged<Thread.Closure>.fromOpaque(arg!)
+// This double declaration is needed becaus of different `pthread_create` parameters requirements, between osx and linux (arm)
+#if os(OSX) || os(iOS) || os(watchOS) || os(tvOS)
+private func ThreadPrivateMain(arg: UnsafeMutableRawPointer) -> UnsafeMutableRawPointer? {
+    let unmanaged = Unmanaged<Thread.Closure>.fromOpaque(arg)
     
     unmanaged.takeUnretainedValue().closure()
     
     unmanaged.release()
     
     return nil
 }
+#elseif os(Linux)
+private func ThreadPrivateMain(arg: UnsafeMutableRawPointer?) -> UnsafeMutableRawPointer? {
```

Comment:
Yes, I had already tried that before but seems that Xcode doesn't like that solution
<img width="880" alt="screenshot 2016-11-02 19 15 21" src="https://cloud.githubusercontent.com/assets/1050414/19941554/d2578c44-a130-11e6-850e-a3d306d22e6b.png">